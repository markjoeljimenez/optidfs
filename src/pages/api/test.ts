import { XMLParser } from 'fast-xml-parser';
import { NextApiRequest, NextApiResponse } from 'next';
import { getToken } from 'next-auth/jwt';

export default async function GET(req: NextApiRequest, res: NextApiResponse) {
	const token = await getToken({ req });

	const response = await fetch(
		`https://fantasysports.yahooapis.com/fantasy/v2/users;use_login=1/games`,
		{
			headers: { Authorization: `Bearer ${token?.accessToken}` },
		}
	);

	res.status(200).json(new XMLParser().parse(await response.text()));
}
