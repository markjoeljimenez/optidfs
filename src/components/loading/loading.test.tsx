import React from 'react';
import { render, screen } from '@/test/render';
import Loading, { LoadingSize, sizeMap, testId } from './loading';

describe('loading', () => {
	it('should match screenshot', () => {
		// Arrange
		const loading = render(<Loading />);

		// Assert
		expect(loading).toMatchSnapshot();
	});

	it('should contain text', () => {
		// Arrange
		const text = 'Loading...';
		render(<Loading text={text} />);

		// Act
		const loading = screen.getByTestId(testId);

		// Assert
		expect(loading).toHaveTextContent(text);
	});

	it('should render large size', () => {
		// Arrange
		render(<Loading size={LoadingSize.Large} />);

		// Act
		const loading = screen.getByTestId(testId);
		const svg = loading.querySelector('svg');

		// Assert
		expect(svg).toHaveClass(sizeMap.get(LoadingSize.Large)!);
	});
});
