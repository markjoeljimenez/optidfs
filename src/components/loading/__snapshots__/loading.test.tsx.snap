// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading should match screenshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center"
        data-testid="loading"
      >
        <svg
          class="animate-spin -ml-1 mr-3 h-4 w-4"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            class="opacity-25"
            cx="12"
            cy="12"
            r="10"
            stroke="currentColor"
            stroke-width="4"
          />
          <path
            class="opacity-75"
            d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center"
      data-testid="loading"
    >
      <svg
        class="animate-spin -ml-1 mr-3 h-4 w-4"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="opacity-25"
          cx="12"
          cy="12"
          r="10"
          stroke="currentColor"
          stroke-width="4"
        />
        <path
          class="opacity-75"
          d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
